### ISSUE TITLE
Improve funnels layout for small screens

### ISSUE THREAD

**Issue by paolodamico** [PostHog/posthog#8990]
## Bug description

The funnels layout seems a bit cumbersome if you don't have a large monitor as the actual funnel visualization takes very little space (even on a 14-inch laptop). It's hard to actually interact with the funnel (I see at most one step) and particularly the table at the bottom.

Because the table at the bottom also contains the links to open the person modal (particularly for drop-offs), it's critical to be able to understand why my users aren't converting, and is not very intuitive that from here I can jump to watch recordings of why users dropped.

<img width="1476" alt="" src="https://user-images.githubusercontent.com/5864173/157924410-bf2aeb37-25a6-4412-a5c2-28ff85da4561.png">

## Expected behavior

I would expect to see at least 3-4 steps without scrolling and have clear indications that I can deep dive into drop-offs to understand why users didn't convert.

## How to reproduce

1. Just create any funnel

## Environment

- [x] PostHog Cloud
- [x] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_

**Comment by paolodamico** [comment]
CC @clarkus

**Comment by clarkus** [comment]
Related https://github.com/PostHog/posthog/issues/8929

**Comment by mariusandra** [comment]
Could we consider switching the funnel mode from "left to right" to the "top to bottom" until we sort this out?

Left to Right:

<img width="1450" alt="Screenshot 2022-03-14 at 20 56 14" src="https://user-images.githubusercontent.com/53387/158250971-cdd73f34-7cfb-4b7c-8361-08695275cbd6.png">

Top to Bottom:

<img width="1463" alt="Screenshot 2022-03-14 at 20 58 23" src="https://user-images.githubusercontent.com/53387/158251277-7274f230-0e14-4524-a930-fc5a4ab3d51b.png">

I wouldn't go so far as to change old saved insights, but new ones could default to the more legible graph type.

**Comment by clarkus** [comment]
We could change the default type, but it doesn't really solve the problem, just makes it less likely to happen. Those layouts vary in a few ways, so it's more about their contents than the flow of information. I don't really think we should change this yet. 

While reviewing I found an easy optimization we could make. At larger widths, we equally size the builder components and the funnel results. We could optimize this by setting a maximum width on builder components (640px usually covers it) and then let the funnel results fill the rest of the available area. I checked on it and it's possible with some breakpoint changes. See the [thread in slack for specific examples.](https://posthog.slack.com/archives/C0368RPHLQH/p1647277703625059)

**Comment by mariusandra** [comment]
Fair point with changing the default type. 

Setting a responsive max-width definitely would already help a bit.

We can tweak the graph itself to save space. We could get all this green space back if we wanted to:

<img width="1435" alt="Screenshot 2022-03-15 at 08 40 12" src="https://user-images.githubusercontent.com/53387/158330224-d551123a-9710-4400-8dcc-d83a88cc7c57.png">

... that's enough to show 2 more funnel steps without changing the experience.

**Comment by clarkus** [comment]
> We can tweak the graph itself to save space. We could get all this green space back if we wanted to:

The column headers won't align with their contents if we make that change. I think it'll make that table be super hard to read. In your example above we're showing a single baseline rate, but once breakdowns are applied, the width of each value can vary greatly. I am working to see if I can optimize this anymore today, but I don't think it'll be as easy and cutting the green space. Working on this first so I should have something pretty soon.

**Comment by mariusandra** [comment]
I mean, obviously, a table should still remain a table after the space is cut, such that headers and cells align. 

The space savings I was imagining are more about for example cutting "👨‍🎨  Completed" down to just "👨‍🎨  C." for the second, third, etc funnel step. Again, just throwing ideas here. Will wait for proposals.

**Comment by clarkus** [comment]
Makes sense. Maybe the table should be decoupled from the vis. The idea was to group corresponding values under the step in the table. That way as you scroll you can see a directly comparison for a given step. I will see what I can do to solve for both.

**Comment by clarkus** [comment]
I didn't make as much progress as I hoped, but I did find some opportunities to optimize the layout to better use the available space. These screens are based on our target viewport size (1280px). In a non-editing context, this gives the funnel results ~1000px of space to render. We can increase the amount of width as the viewport becomes larger or the primary navigation is hidden, etc. [Figma](https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf?node-id=8161:46841#165479568)

Note there are repeated values that don't add up in this example. They are representative of typical character lengths, values, etc. I've tried to optimize for clarity while still showing as much of a step as possible. I've decoupled the table from the left-to-right insight chart. This allows the visualized steps to be a uniform size based on the number of breakdown values included. As the user adds breakdown values, the steps will become wider. Table columns will be determined by the columns shown. By default, the values for each are pretty wide, so we won't show as many steps concurrently here as we do in the chart.

![funnel redraw](https://user-images.githubusercontent.com/254612/158490779-1186e4cd-173e-4556-a46c-e06b02ddad20.png)

This is very much a work in progress but it felt done enough to invite feedback. I will continue working on it and will also show other data states, insight states, etc.
